kv/local_test_cluster.go

```diff
@@ -59,6 +59,14 @@ func (rls *retryableLocalSender) Send(call client.Call) {
 		Tag:         fmt.Sprintf("routing %s locally", call.Method()),
 		MaxAttempts: 2,
 	}
+	// In local tests, the RPCs are not actually sent over the wire. We
+	// need to copy the Txn in order to avoid unexpected sharing between
+	// TxnCoordSender and client.Txn.
+	if header := call.Args.Header(); header.Txn != nil {
+		txn := &proto.Transaction{}
+		*txn = *header.Txn
```

Comment:
This is a shallow copy; will this always be enough? There are some mutable fields in there (e.g. certain_nodes) although I'm not sure if they are mutated in practice. If this is just for tests (i.e. not performance-critical) it might be better to just marshal and unmarshal to guarantee a deep copy.